front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -388,40 +317,73 @@ export function DataSourceViewSelector({
     ]
   );
 
-  const isTableView = viewType === "tables";
+  const isPartiallyChecked = internalIds.length > 0;
+
+  const checkedStatus = selectionConfiguration.isSelectAll
+    ? "checked"
+    : isPartiallyChecked
+      ? "partial"
+      : "unchecked";
 
-  // Show the checkbox by default. Hide it only for tables where no child items are partially checked.
-  const hideCheckbox = readonly || (isTableView && !isPartiallyChecked);
+  const handleSelectAll = () => {
+    document
+      .querySelectorAll<HTMLInputElement>(
+        `#dataSourceViewsSelector-${dataSourceView.dataSource.name} input[type="checkbox"]:first-child`
+      )
+      .forEach((el) => {
+        if (el.checked === isSelectedAll) {
+          el.click();
+        }
+      });
+    setIsSelectedAll(!isSelectedAll);
+  };
 
   return (
-    <Tree.Item
-      key={dataSourceView.dataSource.name}
-      label={getDisplayNameForDataSource(dataSourceView.dataSource)}
-      visual={LogoComponent}
-      type={
-        canBeExpanded(viewType, dataSourceView.dataSource) ? "node" : "leaf"
-      }
-      checkbox={
-        hideCheckbox
-          ? undefined
-          : {
-              checked: checkedStatus,
-              onChange: onToggleSelectAll,
-            }
-      }
-    >
-      <DataSourceViewResourceSelectorTree
-        dataSourceView={dataSourceView}
-        onSelectChange={onSelectChange}
-        owner={owner}
-        parentIsSelected={selectionConfiguration.isSelectAll}
-        readonly={readonly}
-        selectedParents={selectedParents}
-        selectedResourceIds={internalIds}
-        showExpand={config?.isNested ?? true}
-        useContentNodes={useContentNodes}
-        viewType={viewType}
-      />
-    </Tree.Item>
+    <div id={`dataSourceViewsSelector-${dataSourceView.dataSource.name}`}>
+      <Tree.Item
+        key={dataSourceView.dataSource.name}
+        label={getDisplayNameForDataSource(dataSourceView.dataSource)}
+        visual={LogoComponent}
+        type={
+          canBeExpanded(viewType, dataSourceView.dataSource) ? "node" : "leaf"
+        }
+        checkbox={
+          !isRootSelectable
+            ? {
+                checked: checkedStatus,
+                onChange: () => {
+                  selectionConfiguration.isSelectAll =
+                    checkedStatus === "checked";
+                },
```

Comment:
You can't just change a prop, you need to call the updater : 
```suggestion
                onChange: () => {
                  setSelectionConfigurations((prevState) => {
                    const prevSelectionConfiguration =
                      prevState[dataSourceView.sId] ??
                      defaultSelectionConfiguration(dataSourceView);
                    const udpatedConfig = {
                      ...prevSelectionConfiguration,
                      selectedResources: [],
                      isSelectAll: checkedStatus !== "checked",
                    };

                    return {
                      ...prevState,
                      [dataSourceView.sId]: udpatedConfig,
                    };
                  });
                },

```

Reply:
Python's habits 😅